dags/vz_cris_import.py

```diff
@@ -1,31 +1,79 @@
 import os
-from pendulum import datetime
+from pendulum import datetime, duration
 
-from airflow.decorators import dag
+from airflow.models import DAG
 from airflow.operators.docker_operator import DockerOperator
+
+from utils.onepassword import get_env_vars_task
 from utils.slack_operator import task_fail_slack_alert
 
 
-@dag(
-    dag_id="vz-cris-import",
-    description="A process which will import the VZ CRIS data into the database on a daily basis using data on the SFTP endpoint",
-    schedule="0 7 * * *",
-    start_date=datetime(2023, 1, 1, tz="America/Chicago"),
+DEPLOYMENT_ENVIRONMENT = "prod" if os.getenv("ENVIRONMENT") == "production" else "dev"
+
+REQUIRED_SECRETS = {
+    "ENV": {
+        "opitem": "Vision Zero CRIS Import - v2",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.env",
+    },
+    "AWS_ACCESS_KEY_ID": {
+        "opitem": "Vision Zero CRIS Import - v2",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.AWS_ACCESS_KEY_ID",
+    },
+    "AWS_SECRET_ACCESS_KEY": {
+        "opitem": "Vision Zero CRIS Import - v2",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.AWS_SECRET_ACCESS_KEY",
+    },
+    "BUCKET_NAME": {
+        "opitem": "Vision Zero CRIS Import - v2",
+        "opfield": f"common.BUCKET_NAME",
+    },
+    "EXTRACT_PASSWORD": {
+        "opitem": "Vision Zero CRIS Import - v2",
+        "opfield": f"common.EXTRACT_PASSWORD",
+    },
+    "HASURA_GRAPHQL_ENDPOINT": {
+        "opitem": "Vision Zero CRIS Import - v2",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.HASURA_GRAPHQL_ENDPOINT",
+    },
+    "HASURA_GRAPHQL_ADMIN_SECRET": {
+        "opitem": "Vision Zero CRIS Import - v2",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.HASURA_GRAPHQL_ADMIN_SECRET",
+    },
+}
+
+docker_image = f"atddocker/vz-cris-import:{'production' if DEPLOYMENT_ENVIRONMENT == 'prod' else 'latest'}"
+
+
+DEFAULT_ARGS = {
+    "depends_on_past": False,
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 1,
+    "execution_timeout": duration(minutes=60),
```

Comment:
I think Chia's test run cleared the dev extract inbox so I encountered the expected error of `OSError: No extract zips found in S3 bucket`. I noticed that the DAG hangs while it is in `up_for_retry` status. I can't remember if we will see the Slack alert at this point or only when we it truly fails so I wanted to check in case this means we would see a delayed alert.
<img width="303" alt="Screenshot 2024-08-22 at 1 29 53 PM" src="https://github.com/user-attachments/assets/529c766e-0f4f-4aa2-9da1-02240a501dfe">

Reply:
ah yes, great point. iirc we won't see an alert until retries have been expended. perhaps nixing retries altogether is the move. i'd be curious to know, at least in the early going, if retries are even needed 🤔

Reply:
i opted to set `retries` to `0` for now 🙏